cai_causal_graph/time_series_causal_graph.py

```diff
@@ -0,0 +1,755 @@
+        :param node: The node to add.
+        :param kwargs: Additional keyword arguments.
+        :return: The added node.
+        """
+        node = super().add_node(identifier, variable_type=variable_type, meta=meta, node=node, **kwargs)
+        # populate the metadata for each node
+        vname, lag = get_variable_name_and_lag(node.identifier)
+        TimeSeriesCausalGraph._update_node_meta(node, vname, lag)
+        return node
+
+    @reset_attributes
+    def replace_node(
+        self,
+        /,
+        node_id: NodeLike,
+        new_node_id: Optional[NodeLike] = None,
+        *,
+        variable_type: NodeVariableType = NodeVariableType.UNSPECIFIED,
+        meta: Optional[dict] = None,
+    ):
+        """Replace a node in the graph. See `cai_causal_graph.causal_graph.CausalGraph.replace_node` for more details."""
+        super().replace_node(node_id, new_node_id, variable_type=variable_type, meta=meta)
+
+    @reset_attributes
+    def delete_node(self, identifier: NodeLike):
+        """Delete a node from the graph. See `cai_causal_graph.causal_graph.CausalGraph.delete_node` for more details."""
+        super().delete_node(identifier)
+
+    @reset_attributes
+    def delete_edge(self, source: NodeLike, destination: NodeLike):
+        """Delete an edge from the graph. See `cai_causal_graph.causal_graph.CausalGraph.delete_edge` for more details."""
+        super().delete_edge(source, destination)
+
+    @reset_attributes
+    def add_edge(
+        self,
+        /,
+        source: Optional[NodeLike] = None,
+        destination: Optional[NodeLike] = None,
+        *,
+        edge_type: EDGE_T = EDGE_T.DIRECTED_EDGE,
+        meta: Optional[dict] = None,
+        edge: Optional[Edge] = None,
+        **kwargs,
+    ) -> Edge:
+        """
+        Add an edge to the graph. See `cai_causal_graph.causal_graph.CausalGraph.add_edge` for more details.
+
+        In addition to the `CausalGraph.add_edge` method, this method also populates the metadata of the nodes with
+        the variable name and the time lag.
+        """
+        edge = super().add_edge(source, destination, edge_type=edge_type, meta=meta, edge=edge, **kwargs)
+        # populate the metadata for each node
+        for node in self.get_nodes():
+            # extract variable name and lag from node name
+            vname, lag = get_variable_name_and_lag(node.identifier)
+            TimeSeriesCausalGraph._update_node_meta(node, vname, lag)
+
+        return edge
+
+    @classmethod
+    def from_causal_graph(cls, causal_graph: CausalGraph, store_graphs: bool = False) -> TimeSeriesCausalGraph:
+        """
+        Return a time series causal graph from a causal graph.
+
+        This is useful for converting a causal graph from a single time step into a time series causal graph.
+
+        :param causal_graph: The causal graph.
+        :param store_graphs: Whether to store the graphs in the object. Default is False.
```

Comment:
Unclear docstring, don't know what this is doing, what are the graphs? We have only passed one graph

Reply:
It was to store the minimal and summary graphs. I made it simpler and now the system will save them by default.